File: backend/app/schemas/detection.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class PlateReadCreate(BaseModel):
    plate_text: str
    plate_state: str | None = None
    plate_type: str | None = None
    confidence: float
    plate_image_path: str | None = None
    plate_bbox_x: int | None = None
    plate_bbox_y: int | None = None
    plate_bbox_w: int | None = None
    plate_bbox_h: int | None = None
    raw_ocr: str | None = None
    ocr_alternatives: dict = {}


class PlateReadResponse(BaseModel):
    id: uuid.UUID
    plate_text: str
    plate_state: str | None
    plate_type: str | None
    confidence: float
    plate_image_path: str | None
    raw_ocr: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class DetectionCreate(BaseModel):
    session_id: uuid.UUID | None = None
    vehicle_type: str | None = None
    vehicle_color: str | None = None
    vehicle_make: str | None = None
    vehicle_model: str | None = None
    vehicle_year: int | None = None
    vehicle_confidence: float | None = None
    bbox_x: int | None = None
    bbox_y: int | None = None
    bbox_w: int | None = None
    bbox_h: int | None = None
    image_path: str | None = None
    thumbnail_path: str | None = None
    latitude: float | None = None
    longitude: float | None = None
    address: str | None = None
    heading: float | None = None
    speed: float | None = None
    extra: dict = {}
    plate_reads: list[PlateReadCreate] = []


class DetectionResponse(BaseModel):
    id: uuid.UUID
    agent_id: uuid.UUID | None
    session_id: uuid.UUID | None
    vehicle_type: str | None
    vehicle_color: str | None
    vehicle_make: str | None
    vehicle_model: str | None
    vehicle_year: int | None
    vehicle_confidence: float | None
    image_path: str | None
    thumbnail_path: str | None
    latitude: float | None
    longitude: float | None
    address: str | None
    heading: float | None
    speed: float | None
    extra: dict
    created_at: datetime
    plate_reads: list[PlateReadResponse] = []

    model_config = {"from_attributes": True}


class DetectionList(BaseModel):
    items: list[DetectionResponse]
    total: int
    page: int
    page_size: int

File: RepoScanPro/app/src/main/java/com/reposcan/pro/data/model/Detection.kt
package com.reposcan.pro.data.model

import com.google.gson.annotations.SerializedName

data class Detection(
    val id: String,
    @SerializedName("agent_id") val agentId: String?,
    @SerializedName("session_id") val sessionId: String?,
    @SerializedName("vehicle_type") val vehicleType: String?,
    @SerializedName("vehicle_color") val vehicleColor: String?,
    @SerializedName("vehicle_make") val vehicleMake: String?,
    @SerializedName("vehicle_model") val vehicleModel: String?,
    @SerializedName("vehicle_year") val vehicleYear: Int?,
    @SerializedName("vehicle_confidence") val vehicleConfidence: Double?,
    @SerializedName("image_path") val imagePath: String?,
    @SerializedName("thumbnail_path") val thumbnailPath: String?,
    val latitude: Double?,
    val longitude: Double?,
    val address: String?,
    val heading: Double?,
    val speed: Double?,
    val extra: Map<String, Any>?,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("plate_reads") val plateReads: List<PlateRead>
)

data class DetectionList(
    val items: List<Detection>,
    val total: Int,
    val page: Int,
    @SerializedName("page_size") val pageSize: Int
)
